### PR TITLE
[TUBEMQ-493] reorg restful apis and reorg services

### PR DESCRIPTION
**Pull request by EMsnap**
[TUBEMQ-493] reorg restful apis and reorg services [TUBEMQ-491] rebalance group or consumer [TUBEMQ-496] clean offset in consumer group

### CODE REVIEW

File: tubemq-manager/src/main/java/org/apache/tubemq/manager/controller/group/GroupController.java [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    }
+
+
+    /**
+     * add group to black list for certain topic
+     * @param req
+     * @return
+     * @throws Exception
+     */
+    @PostMapping("/blackGroup/add")
```
Comment (yuanboliu) [inline tubemq-manager/src/main/java/org/apache/tubemq/manager/controller/group/GroupController.java:150]: use method instead

File: tubemq-manager/src/main/java/org/apache/tubemq/manager/controller/group/GroupController.java [hunk 2]
```diff
@@ -53,25 +55,27 @@
 @Slf4j
 public class GroupController {
```
Comment (yuanboliu) [inline tubemq-manager/src/main/java/org/apache/tubemq/manager/controller/group/GroupController.java:56]: missing path mapping for group controller

File: tubemq-manager/src/main/java/org/apache/tubemq/manager/controller/group/GroupController.java [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+            case DELETE:
+                return topicService.deleteOffset(gson.fromJson(req, DeleteOffsetReq.class));
+            default:
+                return TubeMQResult.getErrorResult("no such method");
+        }
+    }
+
+
+
+    @PostMapping("/blackGroup")
```
Comment (yuanboliu) [inline tubemq-manager/src/main/java/org/apache/tubemq/manager/controller/group/GroupController.java:113]: use unified path name, "blackGroup"、"topicConfig" or "black-group"、"topic-config", choose one pattern in all controlles' mapping.

File: tubemq-manager/src/main/java/org/apache/tubemq/manager/service/NodeService.java [hunk 4]
```diff
@@ -449,33 +441,56 @@ public TubeMQResult cloneTopicToBrokers(CloneTopicReq req, NodeEntry master) thr
 
     }
 
-    public TubeMQResult cloneOffsetToOtherGroups(CloneOffsetReq req, NodeEntry master)
-        throws Exception {
 
-        // 1. query the corresponding brokers having given topic
-        TubeHttpTopicInfoList topicInfoList = requestTopicConfigInfo(master, req.getTopicName());
-        TubeMQResult result = new TubeMQResult();
+    private TubeMQResult addBrokersToCluster(AddBrokersReq req, NodeEntry masterEntry) {
```
Comment (yuanboliu) [inline tubemq-manager/src/main/java/org/apache/tubemq/manager/service/NodeService.java:445]: reuse baseRequestMaster method.

File: tubemq-manager/src/main/java/org/apache/tubemq/manager/service/TopicService.java [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+
+
+    public TubeMQResult addConsumer(
+        BatchAddGroupAuthReq req) {
+        NodeEntry nodeEntry =
+            nodeRepository.findNodeEntryByClusterIdIsAndMasterIsTrue(req.getClusterId());
+        if (nodeEntry == null) {
+            return TubeMQResult.getErrorResult("no such cluster");
+        }
+        String url = SCHEMA + nodeEntry.getIp() + ":" + nodeEntry.getWebPort()
```
Comment (yuanboliu) [inline tubemq-manager/src/main/java/org/apache/tubemq/manager/service/TopicService.java:82]: reuse baseRequestMaster

File: tubemq-manager/src/main/java/org/apache/tubemq/manager/service/NodeService.java [hunk 6]
```diff
@@ -21,11 +21,11 @@
 import static org.apache.tubemq.manager.controller.node.request.AddBrokersReq.getAddBrokerReq;
 import static org.apache.tubemq.manager.service.TubeMQHttpConst.ADD_TUBE_TOPIC;
 import static org.apache.tubemq.manager.service.TubeMQHttpConst.BROKER_RUN_STATUS;
+import static org.apache.tubemq.manager.service.TubeMQHttpConst.NO_SUCH_CLUSTER;
 import static org.apache.tubemq.manager.service.TubeMQHttpConst.RELOAD_BROKER;
 import static org.apache.tubemq.manager.service.TubeMQHttpConst.SCHEMA;
-import static org.apache.tubemq.manager.service.TubeMQHttpConst.TOPIC_CONFIG_INFO;
 import static org.apache.tubemq.manager.utils.ConvertUtils.convertReqToQueryStr;
-import static org.apache.tubemq.manager.utils.MasterUtils.*;
+import static org.apache.tubemq.manager.service.MasterService.*;
```
Comment (yuanboliu) [inline tubemq-manager/src/main/java/org/apache/tubemq/manager/service/NodeService.java:28]: avoid using wildcard in import area.

Comment (EMsnap) [reply tubemq-manager/src/main/java/org/apache/tubemq/manager/service/NodeService.java]: done

**Comment by yuanboliu** [comment]
+1 pending on test result

**Pull request merged**